ui/src/createAzureClient.ts

```diff
@@ -120,3 +130,24 @@ class MissingContentTypePolicy extends BaseRequestPolicy {
         return response;
     }
 }
+
+/**
+ * The Azure SDK will only throw errors for bad status codes if it has an "operationSpec", but none of our "generic" requests will have that
+ */
+class StatusCodePolicy extends BaseRequestPolicy {
+    constructor(nextPolicy: RequestPolicy, requestPolicyOptions: RequestPolicyOptions) {
+        super(nextPolicy, requestPolicyOptions);
+    }
+
+    public async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+        const response: HttpOperationResponse = await this._nextPolicy.sendRequest(request);
+        if (response.status < 200 || response.status >= 300) {
```

Comment:
Should we be considering 300 codes as errors?  I thought most 300 codes are all still valid (just redirects) https://developer.att.com/video-optimizer/docs/best-practices/http-300-status-codes

I could see our extension not being able to handle the redirecting part of the response right now though.

Reply:
I mainly did this because that's what ms-rest-js does:
https://github.com/Azure/ms-rest-js/blob/master/lib/policies/deserializationPolicy.ts#L140

I think we could debate what "valid" means, but 200-299 are the only ones listed as "Successful":
https://developer.mozilla.org/docs/Web/HTTP/Status